ext/zend_test/tests/strings_marked_as_utf8.phpt

```diff
@@ -0,0 +1,150 @@
+--TEST--
+Check that strings are marked as valid UTF-8
+--EXTENSIONS--
+zend_test
+--FILE--
+<?php
+echo "Empty strings:\n";
+$s = "";
+var_dump(zend_test_is_string_marked_as_valid_utf8($s));
+
+echo "Known strings:\n";
+$s = "c";
+var_dump(zend_test_is_string_marked_as_valid_utf8($s));
+
+echo "Integer cast to string:\n";
+$i = 2563;
+$s = (string) $i;
+var_dump($s);
+var_dump(zend_test_is_string_marked_as_valid_utf8($s));
+
+echo "Float cast to string:\n";
+$f = 26.7;
+$s = (string) $f;
+var_dump($s);
+var_dump(zend_test_is_string_marked_as_valid_utf8($s));
+$f = 2e100;
+$s = (string) $f;
+var_dump($s);
+var_dump(zend_test_is_string_marked_as_valid_utf8($s));
+
+echo "Concatenation known valid UTF-8 strings in variables:\n";
+$s1 = "f";
+$s2 = "o";
+$s = $s1 . $s2;
+var_dump($s);
+var_dump(zend_test_is_string_marked_as_valid_utf8($s));
```

Comment:
after each `$s` is asserted, it would be good to perform the same op with it with invalid UTF-8 and assert the new string is really invalid

Reply:
Very good idea.

Reply:
Ooh, I see @Girgias put that in a separate test file.